Fargate

ferruzzi opened the pull request:
Added EKS support for Fargate Profiles with unit testing:
- list_fargate_profiles
- create_fargate_profile
- describe_fargate_profile
- delete_fargate_profile

bblommers commented on moto/eks/models.py:
```diff
@@ -498,6 +675,51 @@ def validate_launch_template_combination(disk_size, remote_access):
+
+
+def generate_uuid():
+    return "-".join([random_string(_) for _ in [8, 4, 4, 4, 12]]).lower()
```
Can we use the build-in method here?
```
from uuid import uuid4
return str(uuid4())
```

bblommers commented on tests/test_eks/test_eks.py:
```diff
@@ -797,6 +847,490 @@ def test_create_nodegroup_handles_launch_template_combinations(
+    )
+
+    if settings.TEST_SERVER_MODE:
+        RegExTemplates.ISO8601_FORMAT.match(result_time).should.be.true
```
The sure-library has a match-method, that we can use AFAIK:
```
RegExTemplates.ISO8601_FORMAT.should.match(result_time)
```
Slightly easier to read.

ferruzzi replied:
Neat, I didn't know that one.  Thanks!

ferruzzi replied:
Looks like the server mode tests don't like that:  https://github.com/spulec/moto/pull/4185/checks?check_run_id=3364472186#step:9:4609

bblommers replied:
Mea culpa! I honestly thought that would work

bblommers commented on tests/test_eks/test_eks_constants.py:
```diff
@@ -49,6 +51,9 @@
 NODEROLE_ARN_VALUE = "arn:aws:iam::123456789012:role/role_name"
 
+POD_EXECUTION_ROLE_ARN_KEY = "podExecutionRoleArn"
+POD_EXECUTION_ROLE_ARN_VALUE = "arn:aws:iam::123456789012:role/role_name"
```
The account ID is centralized, so we should use that over hardcoded values where possible:
```
from moto.core import ACCOUNT_ID
```
(Same for `NODEROLE_ARN_VALUE `, actually - I must have missed that in the last review)

bblommers reviewed (changes requested):
Hi @ferruzzi, thanks for the PR!
You might have to rebase, there's a merge conflict that's stopping the CI from running

ferruzzi commented:
Sorry about that; you are right, I did forget to rebase.  Rebased, which required a force push, and added pr fixes commit.

FYI:  It looks like [this commit](https://github.com/spulec/moto/commit/76094f012bfbd90f583ef47e0ce313f631f00742) broke py2.7 compatibility with a syntax error at `moto/core/models.py::548`

bblommers reviewed (approved):
LGTM! Thanks again @ferruzzi

ferruzzi commented:
I'm going to be taking some offline time, but afterward I'd like to fix up the EKS tests I contributed previously; I've learned better test-writing practices since that contrib so I'll clean those up and add some support for the EKS waiters I think.  We'll see.  Till then... take care 😁

bblommers commented:
Awesome - tech debt PR's are always welcome 👍
